hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieDeleteBlock.java

```diff
@@ -86,15 +88,25 @@ public HoodieDeleteBlock(Option<byte[]> content, FSDataInputStream inputStream,
         int dataLength = dis.readInt();
         byte[] data = new byte[dataLength];
         dis.readFully(data);
-        this.keysToDelete = SerializationUtils.<HoodieKey[]>deserialize(data);
+        this.recordsToDelete = deserialize(version, data);
         deflate();
       }
-      return keysToDelete;
+      return recordsToDelete;
     } catch (IOException io) {
       throw new HoodieIOException("Unable to generate keys to delete from block content", io);
     }
   }
 
+  private static DeleteRecord[] deserialize(int version, byte[] data) {
+    if (version == 1) {
+      // legacy version
+      HoodieKey[] keys = SerializationUtils.<HoodieKey[]>deserialize(data);
```

Comment:
I think adding log to tell the user that there is an delete record of old version, it may cause data loss.